### ISSUE TITLE
ChangeLog.md open by change-log-mode

### ISSUE THREAD

**Issue by ncaq** [jrblevin/markdown-mode#331]
I open ChangeLog.md, then major-mode is change-log-mode else markdown-mode.

This problem relation to #127.
I want to remove `append` of `add-to-list` arguments.

## Expected Behavior

Emacs use markdown-mode.

## Actual Behavior

Emacs use change-log-mode.

## Steps to Reproduce

1. open `ChangeLog.md`

## Software Versions

- Markdown Mode:  2.4-dev
- Emacs: 25.3.1
- OS: Gentoo Linux

**Comment by ncaq** [comment]
I wrote workaround over an hour.

~~~el
(cl-delete-if (lambda (element) (equal (cdr element) 'markdown-mode)) auto-mode-alist)
(add-to-list 'auto-mode-alist '("\\.md\\'" . markdown-mode))
(add-to-list 'auto-mode-alist '("\\.markdown\\'" . markdown-mode))
~~~

`auto-mode-alist` is diffcult…
I think good to markdown-mode add-to-list to first or do not add-to-list.
I want to `add-to-list` to first.

`auto-mode-alist` prefer asc element.
So, People who prefer gfm-mode should add elements to `auto-mode-alist`.
But, `add-to-list` don't add perfect equal element.
And it don't reorder list.
So, I most use `cl-remove-if` function.

I did not create pull request because `add-to-list` append argument is your change.
